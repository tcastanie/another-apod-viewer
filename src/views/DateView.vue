<template>
  <ApodViewer :date-param="$route.params.date" />
</template>

<script>
import { useStore } from "@/stores/apod";
import ApodViewer from "../components/ApodViewer.vue";
export default {
  components: { ApodViewer },
  setup() {
    const store = useStore();
    return { store };
  },
};
</script>
