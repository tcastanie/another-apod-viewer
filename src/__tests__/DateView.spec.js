import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import { routes } from "@/router/index.js";
import { createTestingPinia } from "@pinia/testing";
import DateView from "@/views/DateView.vue";

let router;
beforeEach(async () => {
  router = createRouter({
    history: createWebHistory(),
    routes: routes,
  });
});

const today = new Date();
const todayDate = today.toISOString().split("T")[0];
const yesterday = new Date();
yesterday.setDate(yesterday.getDate() - 1);
const yesterdayDate = yesterday.toISOString().split("T")[0];
const tomorrow = new Date();
tomorrow.setDate(tomorrow.getDate() + 1);
const tomorrowDate = tomorrow.toISOString().split("T")[0];

const testRoute = `/${todayDate}`;
describe("DateView", async () => {
  it("renders properly", async () => {
    router.push(testRoute);
    await router.isReady();
    const wrapper = mount(DateView, {
      global: {
        plugins: [router, createTestingPinia()],
        stubs: {
          "apod-viewer": true,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("have correct links", async () => {
    router.push(testRoute);
    await router.isReady();
    const wrapper = mount(DateView, {
      global: {
        plugins: [router, createTestingPinia()],
        stubs: {
          "apod-viewer": true,
        },
      },
    });
    expect(wrapper.vm.linkPreviousDay).toContain(yesterdayDate);
    expect(wrapper.findAll("a")[0].attributes("href")).toContain(yesterdayDate);
    expect(wrapper.vm.linkNextDay).toContain(tomorrowDate);
  });

  it("displays correct links", async () => {
    router.push(testRoute);
    await router.isReady();
    const wrapper = mount(DateView, {
      global: {
        plugins: [router, createTestingPinia()],
        stubs: {
          "apod-viewer": true,
        },
      },
    });
    expect(wrapper.findAll("a")[0].attributes("href")).toContain(yesterdayDate);
    expect(wrapper.vm.isNextDayValid).toBe(false);
  });
});
