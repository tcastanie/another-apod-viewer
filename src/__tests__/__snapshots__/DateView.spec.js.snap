// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DateView > renders properly 1`] = `
"<apod-viewer-stub data-v-464d9542=\\"\\" dateparam=\\"2023-08-07\\"></apod-viewer-stub>
<div data-v-464d9542=\\"\\" class=\\"previous-next\\"><a data-v-464d9542=\\"\\" href=\\"/2023-08-06\\" class=\\"\\">Previous</a>
  <!--v-if-->
</div>"
`;
